diff --git a/clade/extensions/callgraph.py b/clade/extensions/callgraph.py
index 51ddeb7..f15cecf 100644
--- a/clade/extensions/callgraph.py
+++ b/clade/extensions/callgraph.py
@@ -116,8 +116,8 @@ class Callgraph(Extension):
                     (f for f in possible_files if self._t_unit_is_common(f, context_file)),
                     (f for f in possible_files if self.funcs[func][f]["type"] == "exported") if call_type == "global" else tuple(),
                     (f for f in possible_files if self._files_are_linked(f, context_file) and
-                        any(self._t_unit_is_common(cf, context_file) for cf in self.funcs[func][f]["declarations"])) if call_type == "global" else tuple(),
-                    (f for f in possible_files if any(self._t_unit_is_common(cf, context_file) for cf in self.funcs[func][f]["declarations"]))
+                        any(self._t_unit_is_common(cf, context_file) for cf in self.funcs[func][f]["declarations"] or [])) if call_type == "global" else tuple(),
+                    (f for f in possible_files if any(self._t_unit_is_common(cf, context_file) for cf in self.funcs[func][f]["declarations"] or []))
                     if call_type == "global" else tuple(),
                     ['unknown']):
                 matched_files = tuple(files)
