Index: scripts/benchmark/vcloud.py
===================================================================
--- scripts/benchmark/vcloud.py	(revision 29088)
+++ scripts/benchmark/vcloud.py	(working copy)
@@ -94,6 +94,10 @@
         else:
             logLevel = "INFO"
         heapSize = benchmark.config.cloudClientHeap + numberOfRuns//10 # 100 MB and 100 kB per run
+        if heapSize < 256:
+            heapSize = 256
+        if heapSize > 1024:
+            heapSize = 1024
         lib = os.path.join(_ROOT_DIR, "lib", "java-benchmark", "vcloud.jar")
         cmdLine = ["java", "-Xmx"+str(heapSize)+"m", "-jar", lib, "benchmark", "--loglevel", logLevel, \
                    "--run-collection-meta-information", meta_information, \
--- src/org/sosy_lab/cpachecker/util/predicates/pathformula/SSAMap.java	(revision 29066)
+++ src/org/sosy_lab/cpachecker/util/predicates/pathformula/SSAMap.java	(working copy)
@@ -62,14 +61,12 @@
       new MergeConflictHandler<String, CType>() {
         @Override
         public CType resolveConflict(String name, CType type1, CType type2) {
-          Preconditions.checkArgument(
-              (type1 instanceof CFunctionType && type2 instanceof CFunctionType)
-                  || (isEnumPointerType(type1) && isEnumPointerType(type2))
-                  || type1.equals(type2),
-              "Cannot change type of variable %s in SSAMap from %s to %s",
-              name,
-              type1,
-              type2);
+          /*
+           * Preconditions.checkArgument( (type1 instanceof CFunctionType && type2 instanceof
+           * CFunctionType) || (isEnumPointerType(type1) && isEnumPointerType(type2)) ||
+           * type1.equals(type2), "Cannot change type of variable %s in SSAMap from %s to %s", name,
+           * type1, type2);
+           */

           return type1;
         }
Index: src/org/sosy_lab/cpachecker/util/refinement/UseDefRelation.java
===================================================================
--- src/org/sosy_lab/cpachecker/util/refinement/UseDefRelation.java	(revision 27049)
+++ src/org/sosy_lab/cpachecker/util/refinement/UseDefRelation.java	(working copy)
@@ -427,7 +425,7 @@
     final Set<ASimpleDeclaration> allLeftHandSideVariables = acceptAll(leftHandSide);
     final Set<ASimpleDeclaration> leftHandSideUses = new HashSet<>(filter(allLeftHandSideVariables, not(in(assignedVariables))));

-    if(assignedVariables.size() > 1) {
+    if(assignedVariables.size() > 1 || assignedVariables.size() == 0) {
       return;
     }
Index: src/org/sosy_lab/cpachecker/cfa/CFASecondPassBuilder.java
===================================================================
--- src/org/sosy_lab/cpachecker/cfa/CFASecondPassBuilder.java	(revision 29088)
+++ src/org/sosy_lab/cpachecker/cfa/CFASecondPassBuilder.java	(working copy)
@@ -191,9 +191,10 @@
             + actualParameters + " parameter(s)", edge);

       case C:
-        throw new CParserException("Method " + functionName + " takes "
+        return;
+        /*throw new CParserException("Method " + functionName + " takes "
             + declaredParameters + " parameter(s) but is called with "
-            + actualParameters + " parameter(s)", edge);
+            + actualParameters + " parameter(s)", edge);*/

       default:
         throw new AssertionError("Unhandled language " + language);
Index: src/org/sosy_lab/cpachecker/cfa/parser/eclipse/c/CFABuilder.java
===================================================================
--- src/org/sosy_lab/cpachecker/cfa/parser/eclipse/c/CFABuilder.java	(revision 29088)
+++ src/org/sosy_lab/cpachecker/cfa/parser/eclipse/c/CFABuilder.java	(working copy)
@@ -387,8 +387,7 @@
     String functionName = startNode.getFunctionName();

     if (cfas.containsKey(functionName)) {
-      throw new CFAGenerationRuntimeException("Duplicate function " + functionName
-          + " in " + startNode.getFileLocation() + " and " + cfas.get(functionName).getFileLocation());
+      return;
     }
     cfas.put(functionName, startNode);
     cfaNodes.putAll(functionName, functionBuilder.getCfaNodes());

Index: src/org/sosy_lab/cpachecker/cpa/predicate/BAMPredicateAbstractionRefinementStrategy.java
===================================================================
--- src/org/sosy_lab/cpachecker/cpa/predicate/BAMPredicateAbstractionRefinementStrategy.java	(revision 29687)
+++ src/org/sosy_lab/cpachecker/cpa/predicate/BAMPredicateAbstractionRefinementStrategy.java	(working copy)
@@ -85,6 +85,7 @@
     // (with 'aggressive caching').
     // -- to have an imprecise predicate-reduce-operator, which can be refined.

+    pRepeatedCounterexample = false;
     // use flags to wait for the second repeated CEX
     if (!pRepeatedCounterexample) {
       pRepeatedCounterexample = false;
Index: src/org/sosy_lab/cpachecker/cpa/automaton/AutomatonExpressionArguments.java
===================================================================
--- src/org/sosy_lab/cpachecker/cpa/automaton/AutomatonExpressionArguments.java	(revision 28904)
+++ src/org/sosy_lab/cpachecker/cpa/automaton/AutomatonExpressionArguments.java	(working copy)
@@ -24,6 +24,7 @@
 package org.sosy_lab.cpachecker.cpa.automaton;
 
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
 import java.math.BigInteger;
 import java.util.Collections;
 import java.util.HashMap;
@@ -60,6 +61,8 @@
   private AutomatonState state;
   private CFAEdge cfaEdge;
   private LogManager logger;
+  private String curArgSign = null;
+  private static Map<AAstNode, String> cache = Maps.newHashMap();
 
   /**
    * In this String all print messages of the Transition are collected.
@@ -161,7 +164,7 @@
           this.getLogger().log(Level.WARNING, "could not replace the transition variable $" + varKey + " (not found).");
           return null;
         } else {
-          result.append(var.toASTString());
+          result.append(process(var));
         }
       } catch (NumberFormatException e) {
         this.getLogger().log(Level.WARNING, "could not parse the int in " + matcher.group() + " , leaving it untouched");
@@ -189,6 +192,52 @@
     return result.toString();
   }
 
+  private CAstNode findSubstituteArgSign(CAstNode pNode) {
+    // System.out.println(pNode + " : " + pNode.getClass());
+    if (pNode instanceof CIdExpression) {
+      curArgSign = ((CIdExpression) pNode).getName();
+      return null;
+    } else if (pNode instanceof CArraySubscriptExpression) {
+      // Take value of automata set variables in CArraySubscriptExpression.
+      CArraySubscriptExpression expr = (CArraySubscriptExpression) pNode;
+      return findSubstituteArgSign(expr.getArrayExpression());
+    } else if (pNode instanceof CFieldReference) {
+      // Execute operations for automata variables, which are encoded in field reference.
+      CFieldReference expr = (CFieldReference) pNode;
+      String argSign =
+          expr.getFieldName() + "_of_" + expr.getFieldOwner().getExpressionType().toString();
+      argSign = argSign.replace("(", "").replace(")", "").replace("*", "").replace(" ", "_");
+      curArgSign = argSign;
+      return null;
+      // return new CStringLiteralExpression(pNode.getFileLocation(), CNumericTypes.INT, argSign);
+    }
+    // Do not substitute pNode.
+    return null;
+  }
+
+  private String process(AAstNode var) {
+    if (cache.containsKey(var)) {
+      return cache.get(var);
+    }
+    SubstitutingCAstNodeVisitor visitor =
+        new SubstitutingCAstNodeVisitor(this::findSubstituteArgSign);
+    String result;
+    if (var instanceof CExpression) {
+      curArgSign = null;
+      CExpression substitutedExpr = (CExpression) ((CExpression) var).accept(visitor);
+      if (curArgSign != null) {
+        result = curArgSign;
+      } else {
+        result = substitutedExpr.toASTString();
+      }
+    } else {
+      result = var.toASTString();
+    }
+    // System.out.println("Result " + result + " for " + var);
+    cache.put(var, result);
+    return result;
+  }
+
   public AutomatonState getState() {
     return state;
   }
@@ -252,7 +301,7 @@
           String name = subscriptExpr;
           CAstNode transitionVariable = getTransitionVariable(subscriptExpr);
           if (transitionVariable != null) {
-            name = transitionVariable.toASTString();
+            name = process(transitionVariable);
           }
           return new CIntegerLiteralExpression(
               pNode.getFileLocation(),
